File: mullins-roger/scramblies.js
function scramble(str1, str2) {

    var counter = {},
        keys = 0;

    for (let i = 0; i < str2.length; i++) {
        if (!counter[str2[i]]) {
            counter[str2[i]] = 0;
            keys++;
        }
        counter[str2[i]]++;
    }

    for (let i = 0; i < str1.length; i++) {
        if (!counter[str1[i]]) continue;
        if (!--counter[str1[i]] && !--keys) return true;
    }
    return false;
}

File: estus-zachary/scramblies.py
function scramble(str1, str2){
  const reducer = (container, element) => {
    if(container[element] === undefined) {
      container[element] = 1;
      return container;
    } else {
      container[element] += 1;
      return container;
    }
  }
  
  var container1 = str1.split('').reduce(reducer,{});
  
  var container2 = str2.split('').reduce(reducer,{});
  
  for (var prop in container2) {
    if(!container1[prop]) return false;
    
    if(container1[prop] < container2[prop]) return false;
  }
  return true;
}
